### PR TITLE
[DOCS-6430] Update ASM Serverless compatibility matrix

### PR DESCRIPTION
**Pull request by sourabhkatti**
### What does this PR do? What is the motivation?

This PR updates the ASM compatibility page for Serverless with removing beta for AWS Lambda, and adding support for Google Cloud Run and Azure App Service.

### Merge instructions

- [x] Please merge with [20209](https://github.com/DataDog/documentation/pull/20209) after reviewing

### CODE REVIEW

File: content/en/security/application_security/enabling/compatibility/serverless.md [hunk 1]
```diff
@@ -6,20 +6,38 @@ type: multi-code-lang
...
+- API Gateway v1 (Rest API)
+- API Gateway v2 (HTTP API)
+- Function URL
 
-### Version dependencies
+<div class="alert alert-info">If you would like to see support added for any of the unsupported capabilities, let us know! Fill out <a href="https://forms.gle/gHrxGQMEnAobukfn7">this short form to send details</a>.</div>
 
-- Lambda Extension version: `39`
-- Serverless plugin version: `5.20.0`
+Get started with ASM monitored on your services →
```
Comment (RomainMuller) [inline content/en/security/application_security/enabling/compatibility/serverless.md:38]:
```suggestion
Get started with ASM monitoring on your services →
```

File: content/en/security/application_security/enabling/compatibility/serverless.md [hunk 2]
```diff
@@ -37,33 +55,44 @@ Go
...
+| Java  		| beta         	| 	                			|						|
+| .NET    	| beta         	|  	                          		|						|
+| Node 		| beta     		| beta	              			|						|
+| Python   	| beta         	| beta                 			|						|
+| Ruby   	| beta         	|  	                 			|						|
+| PHP   	| 		        	|	            			|						|
+| Go   		| beta         	| 	               			|						|
 
-The following ASM capabilities are supported for serverless, for the specified Datadog Lambda extension version:
+Get started with ASM monitored on your services →
```
Comment (RomainMuller) [inline content/en/security/application_security/enabling/compatibility/serverless.md:74]:
```suggestion
Get started with ASM monitoring on your services →
```

File: content/en/security/application_security/enabling/compatibility/serverless.md (hunk 1 above)
Comment (aliciascott) [reply content/en/security/application_security/enabling/compatibility/serverless.md]: I elected just to remove this sentence as it seemed redundant

File: content/en/security/application_security/enabling/compatibility/serverless.md (hunk 2 above)
Comment (aliciascott) [reply content/en/security/application_security/enabling/compatibility/serverless.md]: same with this one, removed this sentence here

**Review by Julio-Guerra — APPROVED** [review]

**Review by RomainMuller — APPROVED** [review]

File: content/en/security/application_security/enabling/compatibility/serverless.md [hunk 3]
```diff
@@ -6,20 +6,34 @@ type: multi-code-lang
...
+| Java  		| {{< X >}}         	| 	                			|						|
+| .NET    	| {{< X >}}         	|  	                          		|						|
+| Node 		| {{< X >}}     		| beta	              			|						|
+| Python   	| {{< X >}}         	| beta                 			|						|
+| Ruby   	| {{< X >}}         	|  	                 			|						|
+| PHP   	| 		        	|	            			|						|
+| Go   		| {{< X >}}         	| 	               			|						|
+
+### Supported trigger types
+ASM Threat Detection supports HTTP requests as function input only, as it is the highest likelihood of attackers exploiting a serverless application. HTTP requests typically come from AWS services such as:
```
Comment (urseberry) [inline content/en/security/application_security/enabling/compatibility/serverless.md:27]:
```suggestion
ASM Threat Detection supports HTTP requests as function input only, as that channel has the highest likelihood of attackers exploiting a serverless application. HTTP requests typically come from AWS services such as:
```

File: content/en/security/application_security/enabling/compatibility/serverless.md [hunk 4]
```diff
@@ -6,20 +6,34 @@ type: multi-code-lang
...
+- Application Load Balancer (ALB)
+- API Gateway v1 (Rest API)
+- API Gateway v2 (HTTP API)
+- Function URL
 
-### Version dependencies
+<div class="alert alert-info">If you would like to see support added for any of the unsupported capabilities, let us know! Fill out <a href="https://forms.gle/gHrxGQMEnAobukfn7">this short form to send details</a>.</div>
```
Comment (urseberry) [inline content/en/security/application_security/enabling/compatibility/serverless.md:33]:
```suggestion
<div class="alert alert-info">If you would like to see support added for any of the unsupported capabilities, fill out this <a href="https://forms.gle/gHrxGQMEnAobukfn7">form</a> to send feedback.</div>
```

File: content/en/security/application_security/enabling/compatibility/serverless.md [hunk 5]
```diff
@@ -37,36 +51,43 @@ Go
...
+| PHP   	| 		        	|	            			|						|
+| Go   		| beta         	| 	               			|						|
 
-<div class="alert alert-info">If you would like to see support added for any of the unsupported capabilities, let us know! Fill out <a href="https://forms.gle/gHrxGQMEnAobukfn7">this short form to send details</a>.</div>
 
-## Supported trigger types
+## Azure App Service
 
-ASM Threat Detection supports HTTP requests as function input only. These typically come from AWS services such as:
+Only *web applications* are supported, Azure Functions are not supported with ASM.
```
Comment (urseberry) [inline content/en/security/application_security/enabling/compatibility/serverless.md:74]:
```suggestion
Only *web applications* are supported. Azure Functions are not supported with ASM.
```

**Review by aliciascott — APPROVED** [review]

**Review by jhgilbert — APPROVED** [review]

**Pull request merged**